src/java.base/unix/native/libnio/fs/UnixNativeDispatcher.c

```diff
@@ -687,7 +687,7 @@ Java_sun_nio_fs_UnixNativeDispatcher_stat0(JNIEnv* env, jclass this,
         if (err == 0) {
             copy_statx_attributes(env, &statx_buf, attrs);
             return 0;
-        } else {
+        } else if (errno != ENOSYS && errno != EPERM) {
```

Comment:
I'd prefer if we only handled `EPERM` with a comment about the docker version issue. `ENOSYS` shouldn't happen since we test for `statx` availability in init() [here](https://github.com/openjdk/jdk21u-dev/blob/bd5b120f516db77c12233d53c8b834dda5b82a89/src/java.base/unix/native/libnio/fs/UnixNativeDispatcher.c#L432-L435). Also, we are in the `my_statx_func != NULL` branch.